lib/vanagon/engine/local.rb

```diff
@@ -1,40 +1,38 @@
+require 'vanagon/engine/base'
 require 'vanagon/utilities'
 require 'vanagon/errors'
-require 'benchmark'
 
 class Vanagon
   class Engine
-    class Local
-      attr_accessor :target
+    class Local < Base
 
       def initialize(platform, target = nil)
         @platform = platform
         @target = "local machine"
         @name = 'local'
+        super
+        @required_attributes = Array.new
       end
 
       # Dispatches the command for execution
-      def dispatch(command)
-        puts Benchmark.measure { local_command(command, @workdir) }
-      end
-
-      # Steps needed to tear down or clean up the system after the build is
-      # complete
-      def teardown
-      end
-
-      # This method will take care of validation and target selection all at
-      # once as an easy shorthand to call from the driver
-      def startup(workdir)
-        @workdir = workdir
-        script = @platform.provisioning.join(' && ')
-        dispatch(script)
+      def dispatch(command, return_output = false)
+        # unset environment variables set by 'bundle exec build'
```

Comment:
@MikaelSmith I think we might be able to add a dependency on `bundler-unload` and use that to "unbundle" a little more safely than directly altering state.

```ruby
>> require 'bundler-unload'
=> true
>> Bundler.load
>> ENV.select { |k,_| %w(BUNDLE_BIN_PATH BUNDLE_GEMFILE GEM_HOME GEM_PATH RUBYOPT RUBYLIB).include? k }
=> {
           "GEM_HOME" => "/Users/mckern/Documents/Repositories/bundlertest/.bundle/gems/ruby/2.1.0",
           "GEM_PATH" => "",
            "RUBYOPT" => "-rbundler/setup",
            "RUBYLIB" => "/usr/local/rvm/gems/ruby-2.1.8@global/gems/bundler-1.11.2/lib",
    "BUNDLE_BIN_PATH" => "/usr/local/rvm/gems/ruby-2.1.8@global/gems/bundler-1.11.2/exe/bundle",
     "BUNDLE_GEMFILE" => "/Users/mckern/Documents/Repositories/bundlertest/Gemfile"
}
>> all_gems = Bundler.rubygems.plain_specs
true
>> Bundler.unload!(all_gems)
true
>> ENV.select { |k,_| %w(BUNDLE_BIN_PATH BUNDLE_GEMFILE GEM_HOME GEM_PATH RUBYOPT RUBYLIB).include? k }
=> {
    "GEM_HOME" => "/usr/local/rvm/gems/ruby-2.1.8",
    "GEM_PATH" => "/usr/local/rvm/gems/ruby-2.1.8:/usr/local/rvm/gems/ruby-2.1.8@global",
     "RUBYOPT" => "-rbundler/setup",
     "RUBYLIB" => "/usr/local/rvm/gems/ruby-2.1.8@global/gems/bundler-1.11.2/lib"
}
```